optuna/terminator/improvement/evaluator.py

```diff
@@ -60,59 +83,135 @@ class RegretBoundEvaluator(BaseImprovementEvaluator):
             A minimum number of additional samples to estimate the lower confidence bound.
             Default to 2000.
-    """
+
+    For further information about this evaluator, please refer to the following paper:
+
+    - `Automatic termination for hyperparameter optimization <https://proceedings.mlr.press/v188/makarova22a.html>`_
+    """  # NOQA: E501
 
     def __init__(
         self,
-        gp: Optional[BaseGaussianProcess] = None,
         top_trials_ratio: float = DEFAULT_TOP_TRIALS_RATIO,
         min_n_trials: int = DEFAULT_MIN_N_TRIALS,
-        min_lcb_n_additional_samples: int = 2000,
+        seed: int | None = None,
     ) -> None:
-        self._gp = gp or _BoTorchGaussianProcess()
         self._top_trials_ratio = top_trials_ratio
         self._min_n_trials = min_n_trials
-        self._min_lcb_n_additional_samples = min_lcb_n_additional_samples
-
-    def get_preprocessing(self, add_random_inputs: bool = False) -> BasePreprocessing:
-        processes = [
-            SelectTopTrials(
-                top_trials_ratio=self._top_trials_ratio,
-                min_n_trials=self._min_n_trials,
-            ),
-            UnscaleLog(),
-            ToMinimize(),
-        ]
-
-        if add_random_inputs:
-            processes += [AddRandomInputs(self._min_lcb_n_additional_samples)]
+        self._log_prior = prior.default_log_prior
+        self._minimum_noise = prior.DEFAULT_MINIMUM_NOISE_VAR
+        self._optimize_n_samples = 2048
+        self._rng = LazyRandomState(seed)
+
+    def _get_top_n(
+        self, normalized_params: np.ndarray, values: np.ndarray
+    ) -> tuple[np.ndarray, np.ndarray]:
+        assert len(normalized_params) == len(values)
+        n_trials = len(normalized_params)
+        top_n = np.clip(int(n_trials * self._top_trials_ratio), self._min_n_trials, n_trials)
+        top_n_val = np.partition(values, n_trials - top_n)[n_trials - top_n]
+        top_n_mask = values >= top_n_val
+        return normalized_params[top_n_mask], values[top_n_mask]
+
+    def _compute_standardized_regret_bound(
+        self,
+        optuna_search_space: Dict[str, BaseDistribution],
+        kernel_params: gp.KernelParamsTensor,
+        gp_search_space: search_space.SearchSpace,
+        normalized_top_n_params: np.ndarray,
+        standarized_top_n_values: np.ndarray,
+    ) -> float:
+        """
+        In the original paper, f(x) was intended to be minimized, but here we would like to
+        maximize f(x). Hence, the following changes happen:
+            1. min(ucb) over top trials becomes max(lcb) over top trials, and
+            2. min(lcb) over the search space becomes max(ucb) over the search space, and
+            3. Regret bound becomes max(ucb) over the search space minus max(lcb) over top trials.
+        """
+
+        n_params = len(optuna_search_space)
+
+        # calculate max_ucb
+        beta = np.sqrt(_get_beta(n_params, len(normalized_top_n_params)))
```

Comment:
It is not `sqrt_beta` anymore.
```suggestion
        beta = _get_beta(n_params, len(normalized_top_n_params))
```